pyomo/contrib/mindtpy/util.py

```diff
@@ -181,3 +181,84 @@ def var_bound_add(solve_data, config):
                         var.setub(config.integer_var_bound)
                     else:
                         var.setub(config.continuous_var_bound)
+
+
+def generate_Norm2sq_objective_function(model, setpoint_model, discrete_only=False):
+    """Generate objective (FP-NLP subproblem) for minimum euclidean distance to setpoint_model
+    L2 distance of (x,y) = \sqrt{\sum_i (x_i - y_i)^2}
```

Comment:
This documentation needs to be completed and corrected. Remember we are dealing with the squared two-norm.